pallet/nfi/src/lib.rs

```diff
@@ -0,0 +1,381 @@
+// Copyright 2022-2023 Futureverse Corporation Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// You may obtain a copy of the License at the root of this project source code
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+	ensure,
+	pallet_prelude::*,
+	traits::{fungibles::Mutate, tokens::Preservation, Get},
+};
+use frame_system::pallet_prelude::*;
+use seed_pallet_common::{NFIRequest, NFTExt, SFTExt};
+use seed_primitives::{AssetId, Balance, CollectionUuid, SerialNumber, TokenCount, TokenId};
+use sp_runtime::{traits::Zero, DispatchResult, Permill};
+use sp_std::prelude::*;
+
+pub use pallet::*;
+
+mod types;
+use types::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+pub mod mock;
```

Comment:
nit: why exposed to pub api
```suggestion
mod mock;
```

Reply:
Whoops, this was left over after some experiments. Addressed: https://github.com/futureversecom/trn-seed/pull/866/commits/e5e9de4910537d3d4c8f71c18510063a0d9d3f78